### PR TITLE
V1

### PR DESCRIPTION
**Pull request by dfithian**

### CODE REVIEW

File: examples/package.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: example
+version: '1.0'
+maintainer: TVision Insights
+license: AllRightsReserved
```
Comment (mossprescott) [inline examples/package.yaml]: Need a package name, `0.x` version, and license.

File: package.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+name: interpolator
+version: '1.0'
+maintainer: TVision Insights
+license: AllRightsReserved
```
Comment (mossprescott) [inline package.yaml]: Actually, _here_ you need a `0.x` version and a license. Not such a big deal in the example package I guess.

File: src/Data/Interpolation.hs [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+  toJSON = \ case
+    Templated x -> toJSON x
+    Literal x -> toJSON x
+
+maybeGen :: Gen a -> Gen (Maybe a)
+maybeGen x = frequency [(1, pure Nothing), (3, Just <$> x)]
+
+noEnv, noColons :: Gen T.Text
+noEnv = fmap T.pack $ arbitrary `suchThat` (\ s -> not ("_env:" `isPrefixOf` s) && not (null s))
+noColons = fmap T.pack . listOf1 $ arbitrary `suchThat` (\ c -> c /= ':')
```
Comment (mossprescott) [inline src/Data/Interpolation.hs]:
```suggestion
noColons = fmap T.pack . listOf1 $ arbitrary `suchThat` (/= ':')
```

No big deal, obviously (mostly trying out the "suggestion" feature.

File: stack.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+resolver: lts-11.16
```
Comment (mossprescott) [inline stack.yaml]: This is a rather arbitrary LTS version at this point. Maybe do the thing where you include multiple stack.yamls (for this one, whatever the current one is and maybe "latest" however you do that), and build them all in travis. Not sure if it's really that simple, but I've seen it done.

File: examples/package.yaml (hunk 1 above)
Comment (dfithian) [reply examples/package.yaml]: Right, I'm not going to release this, it's just so I know it compiles when I load in GHCi. I will add a license.

File: stack.yaml (hunk 4 above)
Comment (dfithian) [reply stack.yaml]: What I did was override the resolver in travis.yml to also build current LTS and the nightly one.

File: LICENSE [hunk 5]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) [year] [fullname]
```
Comment (mossprescott) [inline LICENSE]:
```suggestion
Copyright (c) 2018 TVision Insights
```

File: package.yaml (hunk 2 above)
Comment (mossprescott) [reply package.yaml]: Seems like packages in Hackage always have 3 or 4 segments, so 0.1.0 or 0.1.0.0? I'm not gonna sweat it though, as long as it starts with `0.`.

**Pull request merged**